lib/beaker/dsl/structure.rb

```diff
@@ -70,6 +70,7 @@ def test_name my_name, &block
       #       'ensure=absent', 'purge=true'))
       #   end
       def teardown &block
+        logger.info('Begin teardown')
```

Comment:
This is a good idea! We might also consider adding an `End teardown` message too.

I admit I haven't tested this, but just looking at the code it seems this is the wrong place for this log message to be printed.  I believe this method is how a test registers a block for callback during the teardown phase -- this isn't actually where/when the test suite is performing teardown.

I think we want to add the log messages before/after we call the blocks, which looks like here: https://github.com/puppetlabs/beaker/blob/master/lib/beaker/test_case.rb#L139-L144